python/lsst/pipe/analysis/utils.py

```diff
@@ -679,13 +680,14 @@ class CentroidDiffErr(CentroidDiff):
     """Functor to calculate difference error for astrometry.
     """
     def __call__(self, catalog):
+        schema = getSchema(catalog)
         firstx = self.first + self.centroid + "_xErr"
         firsty = self.first + self.centroid + "_yErr"
         secondx = self.second + self.centroid + "_xErr"
         secondy = self.second + self.centroid + "_yErr"
 
-        subkeys1 = [catalog.schema[firstx].asKey(), catalog.schema[firsty].asKey()]
-        subkeys2 = [catalog.schema[secondx].asKey(), catalog.schema[secondy].asKey()]
+        subkeys1 = [schema[firstx].asKey(), schema[firsty].asKey()]
+        subkeys2 = [schema[secondx].asKey(), schema[secondy].asKey()]
```

Comment:
OK, I really will jump ahead and look at the `getSchema` function  now, since it clearly doesn't return a list?

Reply:
OK looked at it, and it does return a list, unless it's being passed an AFW table.  So all these places I've marked will only work if the catalogs are afwtables.  I'll continue to mark them, though I suspect this all will get cleared up in other commits.